runtime/src/bank.rs

```diff
@@ -89,15 +89,14 @@ pub struct Bank {
     /// Previous checkpoint of this bank
     parent: Option<Arc<Bank>>,
 
-    /// Hash of the previous checkpoint's state
-    parent_hash: Hash,
+    /// Hash of this Bank's state. Only meaningful after freezing it via `new_from_parent()`.
+    hash: RwLock<Hash>,
```

Comment:
sneaky, RwLock lets us hold RO Arcs to the Bank all over the place?

Reply:
yeah.  Probably too sneaky. It's something I did in the very early days of all the Bank variables.  The best part about it is that it lets you safely operate on different parts of the back from different threads.  It's been good for performance.  The worse part is that we pass `Arc<Bank>` all over the place and you can't tell if the caller is going to modify it without looking at the code.